### PR TITLE
Reconfigure start scripts to use nodemon

### PR DESCRIPTION
**Pull request by gshaw1997**
This PR updates start scripts to use tsc-node and nodemon instead of gulp to start the development server. This fixes the issue gulp and nodemon caused which forced unnecessary nodemon restarts in file change.

### CODE REVIEW

File: src/drivers/express/ExpressAdminRouteDriver.ts [hunk 1]
```diff
@@ -86,18 +80,20 @@ export default class ExpressAdminRouteDriver {
...
       '/learning-objects',
       proxy(LEARNING_OBJECT_SERVICE_URI, {
-      proxyReqPathResolver: req => {
-        console.log(req.body);
-        const route = ADMIN_LEARNING_OBJECT_ROUTES.UPDATE_OBJECT();
-        return route;
-      },
-     }),
+        proxyReqPathResolver: req => {
+          console.log(req.body);
```
Comment (nvisal1) [inline src/drivers/express/ExpressAdminRouteDriver.ts]: Remove `console.log`

File: src/drivers/express/ExpressRouteDriver.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import * as express from 'express';
 import { Router } from 'express';
-import * as proxy from 'express-http-proxy';
-import { ExpressResponder } from '../drivers';
+// tslint:disable-next-line:no-require-imports
+import proxy = require('express-http-proxy');
```
Comment (DonSeannelly) [inline src/drivers/express/ExpressRouteDriver.ts]: Why is this done via a require import?

File: tsconfig.json [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-      "noFallthroughCasesInSwitch": true
-    },
-    "exclude": [
-      "node_modules",
-      "client"
-    ]
-  }
\ No newline at end of file
+  "compilerOptions": {
+    "target": "es2017",
```
Comment (DonSeannelly) [inline tsconfig.json]: What version of node do we need to make sure we have running for all ES 2017 syntax to be available?

Comment (DonSeannelly) [reply tsconfig.json]: I noted this question only inside of this thread, but the question applies to each of the corresponding PRs. Just want to check and make sure the developer machines and our dock or containers are configured correctly. @nvisal1 @gshaw1997

Comment (gshaw1997) [reply tsconfig.json]: These sources state that ES 2017 syntax requires a minimum Node version of 6.5. 
https://node.green/
https://stackoverflow.com/questions/44027764/how-can-i-use-es7-in-nodejs

I believe all of our containers are running a minimum version of Node 8

File: src/drivers/express/ExpressRouteDriver.ts (hunk 2 above)
Comment (gshaw1997) [reply src/drivers/express/ExpressRouteDriver.ts]: :confused: I honestly do not remember. It's been reverted :ok_hand:

**Pull request merged**